PVLV update

rcoreilly opened the pull request:
This is the big PVLV update.  The PVLV part itself is good to go for now (still more to test and tune) but it is not yet integrated and tested with BOA, and it needs just a few more minor things...  anyway, review can start now, and PVLV.md has all the docs updated.

etuleu commented on examples/boa/boa.go:
```diff
@@ -373,26 +370,26 @@ func (ss *Sim) ConfigNet(net *axon.Network) {
 
-	vPgpi.SetRelPos(relpos.Rel{Rel: relpos.RightOf, Other: rew.Name(), YAlign: relpos.Front, Space: space})
-	ach.SetRelPos(relpos.Rel{Rel: relpos.Behind, Other: snc.Name(), XAlign: relpos.Left, Space: space})
+	vPgpi.PlaceRightOf(rew, space)
```
This looks much nicer!

etuleu commented on axon/drivepvlv.go:
```diff
@@ -0,0 +1,534 @@
+
+// DriveVals represents different internal drives, such as hunger, thirst, etc.
+// labels can be provided by specific environments.
+type DriveVals struct {
```
why not use an array?

rcoreilly replied:
no arrays in GPU land..

etuleu commented on axon/deep_net.go:
```diff
@@ -298,9 +298,9 @@ func (nt *Network) AddPTPredLayer(ptMaint, ct, thal *Layer, ptToPredPrjn, ctToPr
-	toThal, fmThal := nt.BidirConnectLayers(ptMaint, thal, toThalPrjn)
-	toThal.SetClass("PredToThal")
-	fmThal.SetClass("ThalToPred")
+	// toThal, fmThal := nt.BidirConnectLayers(ptPred, thal, toThalPrjn)
```
do we want this to stay commented out?

rcoreilly replied:
good point -- just discovered this was being done late last night -- now removed.

rcoreilly replied:
and added comment why

etuleu commented on examples/boa/boa.go:
```diff
@@ -337,16 +340,23 @@ func (ss *Sim) ConfigNet(net *axon.Network) {
 	net.ConnectLayers(effort, alm, full, emer.Forward)
-	net.ConnectLayers(ofcPT, alm, full, emer.Forward)
-	net.ConnectLayers(accPT, alm, full, emer.Forward)
+	// net.ConnectLayers(ofcPT, alm, full, emer.Forward)
```
same here?

etuleu commented on examples/boa/boa.go:
```diff
@@ -576,9 +591,9 @@ func (ss *Sim) ConfigLoops() {
-		man.GetLoop(etime.Test, etime.Trial).Main.Add("NetDataRecord", func() {
-			ss.GUI.NetDataRecord(ss.ViewUpdt.Text)
-		})
+		// man.GetLoop(etime.Test, etime.Trial).Main.Add("NetDataRecord", func() {
```
also here?

etuleu reviewed (approved):
I left just a few comments next to commented out code just so you can take a look again to make sure it's not commented out by mistake. And maybe we can leave a note saying why it is commented out.

I ran it locally also and it looks good! 
Awesome work!

etuleu commented:
Just an idea: maybe we should update the readme file (pvlv.md) with a section about the PTNotMaint layer? I might have missed it so will check it again, but just wanted to write this before I forgot.

rcoreilly commented:
thanks for the comments -- I made another pass through the ConfigNet and removed or commented on stuff that doesn't make sense, and added todo comments for all the stuff that still needs to be tested (later)..

I added the PTNotMaintLayer to DEEP.md (technically part of that framework) and to the boa README.md

etuleu commented on PVLV.md:
```diff
@@ -1,33 +1,188 @@
+
+* multiplicative effort discounting will not produce a net negative on goal failure -- need an extra case for that.
+
+* PT not clearing -- need to clear super too!
```
Is this still an issue?
